src/main/java/org/cloudfoundry/promregator/lite/cfaccessor/SimpleCFAccessorImpl.kt

```diff
@@ -0,0 +1,511 @@
+                } catch (e: UnknownHostException) {
+                    throw ConfigurationException(String.format("The proxy host '%s' cannot be resolved to an IP address; is there a typo in your configuration?", effectiveProxyHost), e)
+                }
+            } else {
+                // the address specified is already an IP address
+                effectiveProxyHost
+            }
+            ProxyConfiguration.builder().host(proxyIP).port(effectiveProxyPort).build()
+        } else {
+            null
+        }
+    }
+
+    private fun cloudFoundryClient(connectionContext: ConnectionContext, tokenProvider: TokenProvider): ReactorCloudFoundryClient {
+        return ReactorCloudFoundryClient.builder().connectionContext(connectionContext).tokenProvider(tokenProvider).build()
+    }
+
+    fun reset(api: String) {
+        val apiConfig = cf.api[api]
+        if(apiConfig != null) {
+            resetCloudFoundryClient(api, apiConfig)
+        }
+    }
+
+    private fun resetCloudFoundryClient(api: String, apiConfig: ApiConfig) {
+        try {
+            val proxyConfiguration = proxyConfiguration(apiConfig)
+            val connectionContext = connectionContext(apiConfig, proxyConfiguration)
+            val tokenProvider = tokenProvider(apiConfig)
+            cloudFoundryClients[api] = cloudFoundryClient(connectionContext, tokenProvider)
+        } catch (e: ConfigurationException) {
+            log.error(e) {"Restarting Cloud Foundry Client failed due to Configuration Exception raised" }
+        }
+    }
+
+//    try {
+//        if (cloudFoundryClient != null) {
+//            /*
+//             * Note: there may still be connections and threads open, which need to be closed.
+//             * https://github.com/promregator/promregator/issues/161 pointed that out.
+//             */
+//            val connectionContext = cloudFoundryClient!!.connectionContext
+//            // Note: connectionContext is ensured to be non-null
+//            if (connectionContext is DefaultConnectionContext) {
+//                /*
+//                 * For the idea see also
+//                 * https://github.com/cloudfoundry/cf-java-client/issues/777 and
+//                 * https://issues.jenkins-ci.org/browse/JENKINS-53136
+//                 */
+//                connectionContext.dispose()
+//            }
+//        }
+//        val proxyConfiguration = proxyConfiguration()
+//        val connectionContext = connectionContext(proxyConfiguration)
+//        val tokenProvider = tokenProvider()
+//        cloudFoundryClient = cloudFoundryClient(connectionContext, tokenProvider)
+//    } catch (e: ConfigurationException) {
+//        log.error("Restarting Cloud Foundry Client failed due to Configuration Exception raised", e)
+//    }
+
+    suspend fun retrieveAllOrgIds(api: String): List<OrganizationResourceV2> {
+        return cfQueryV2<OrganizationResourceV2, ListOrganizationsResponseV2, ListOrganizationsRequestV2>(api, ORG) {
```

Comment:
This is the new paginating mini-DSL. Just need to define a `request` and the `query` and it returns the `List<Resource>`.